<launch>
  <param name="/mini_microphone/chunk" value="512"/>
  <param name="/mini_microphone/length" value="512"/>
  <param name="/mini_microphone/rate" value="44100"/>
  <arg name="cutoff_rate" default="8000"/>
  <arg name="save_image" default="false"/>
  <arg name="save_spectrum" default="false"/>
  <arg name="hitting_target" default="table"/>

  <node pkg="euslib" type="listen_mini_microphone.py" name="listen_mini_microphone" output="screen" respawn="true"/>

  <!-- <node pkg="euslib" type="spectrum_analyzer_mini_microphone.py" name="spectrum_analyzer_mini_microphone" output="screen"/> -->

  <node pkg="euslib" type="save_hitting_sound.py" name="save_hitting_sound" output="screen" respawn="true">
    <param name="cutoff_rate" value="$(arg cutoff_rate)"/>
    <param name="time_to_listen" value="0.3"/>
    <param name="save_image" value="$(arg save_image)"/>
    <param name="save_spectrum" value="$(arg save_spectrum)"/>
    <param name="hitting_target" value="$(arg hitting_target)"/>
  </node>

  <node pkg="image_view" type="image_view" name="sound_image_view">
    <remap from="image" to="/mini_microphone/sound_image"/>
  </node>

  <node pkg="euslib" type="classify_hitting_sound_ros.py" name="classify_hitting_sound" output="screen">
    <param name="cutoff_rate" value="$(arg cutoff_rate)"/>
  </node>

</launch>
